src/bin/pg_autoctl/service_keeper.c

```diff
@@ -321,8 +334,40 @@ keeper_node_active_loop(Keeper *keeper, pid_t start_pid)
 			(void) keeper_call_reload_hooks(keeper, firstLoop, doInit);
 		}
 
-		if (asked_to_stop || asked_to_stop_fast || asked_to_quit)
+		/*
+		 * Upon receiving SIGTERM, start maintenance on the monitor, possibly
+		 * triggering a failover when the primary node is being shut down.
+		 */
+		if (asked_to_stop)
 		{
+			if (keeperState->current_role != MAINTENANCE_STATE &&
```

Comment:
If state is `PREPARE_MAINTENANCE_STATE`, then the user clearly wants maintenance to be the state on startup. So we should handle that state in some special way, probably we want to wait until the transition is completed, but don't want to indicate maintainance should be disabled after boot.

Reply:
Fixed too, as part of the previous comment with SIGINT during a SIGTERM induced shutdown sequence.